clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -171,6 +195,7 @@ public void removeSensor(String name) {
                     }
                 }
             }
```

Comment:
I thought I had a scenario that would make it possible to incorrectly remove a sensor (when its parent is removed). So we just walked through the different scenarios. Say a parent P has child sensors C1, C2.

- Calling record on P would make its scheduled expiration time >= expiration time for C1/C2. This is fine. i.e., C1 and/or C2 may get removed if they don't get updated but that doesn't impact its sibling or the parent.
- Calling record on (say) C1 would cause a record on P as well. So expiration time for P == C1 and that expiration time >= expiration time for C2. This is also good, although there is a small race condition - i.e., if the record on P happens just after it is removed. That remove will cause C1 and C2 to also get removed (when C1 should not be removed).

The expiration time should typically be large (possibly as high as 30 minutes even). It is probably not worth it to optimize the implementation to address the above race condition for that time granularity.